### PR TITLE
Operation: Gnomeregan! [v2]

### LINKED ISSUE #1688: [Event] Operation Gnomeregan (Completely Missing)
##### SMALL DESCRIPTION:

Just like the event Zalazane's Fall (Horde Only Event ID: 61), this is the Alliance version of the pre-cataclysm event. It seems this event doesn't even exist in the AC database. Nor does it exist in https://wowgaming.altervista.org/aowow/?events.

However, this event was definitely there in patch 3.3.5, and I found the event in Warmane's Wotlk DB: http://wotlk.cavernoftime.com/events

##### EXPECTED BLIZZLIKE BEHAVIOUR:

https://wow.gamepedia.com/Operation:_Gnomeregan
https://youtu.be/7saEJx3kEfA

##### CURRENT BEHAVIOUR:

Event doesn't exist at all. Quests are in the DB but I guess it isn't scripted since the entire event is missing.

##### STEPS TO REPRODUCE THE PROBLEM:

1.  No such event exists in the current AzerothCore DB.
2. 
3. 

##### EXTRA NOTES:

I compared https://wowgaming.altervista.org/aowow/?events and http://wotlk.cavernoftime.com/events and begin to wonder is AzerothCore using the most up to date DB for 3.3.5? Even the Event IDs between the 2 databases are different...

##### BRANCH(ES):

master

##### AC HASH/COMMIT:

9eb6e9a

##### OPERATING SYSTEM:

Win 7/10/Svr 2016

##### MODULES:

N/A

##### OTHER CUSTOMIZATIONS:

N/A

### PR DESCRIPTION
**Pull request by FALL1N1**
## CHANGES PROPOSED:
-  Script the entire chain

## ISSUES ADDRESSED:
- Closes #1688

Cherry-Picked: https://github.com/rebirth-core/Rebirth---WoW/commit/4342a1061984330d1e0687cb4e355afcc33248eb
Co-Authored-By: [PatrickFarnkopfOld](https://github.com/rebirth-core/Rebirth---WoW/commits?author=PatrickFarnkopfOld)
Co-Authored-By: [FrEdDy32](https://github.com/FrEdDy32)
Co-Authored-By:[Nebukad](https://github.com/Nebukad)
Co-Authored-By:[rbkai](https://github.com/rbkai)

## TESTS PERFORMED: 
- [Q:25229] A Few Good Gnomes https://youtu.be/-e0h53Y55rI
- [Q:25199] Basic Orders https://youtu.be/uex8-nRpgbw
- [Q:25285] In and Out https://youtu.be/91jFSsTC9U0
- [Q:25289] One Step Forward... https://youtu.be/pzJbTsxhmwU
- [Q:25295] Press Fire https://youtu.be/b_DAifd4uxo
- [Q:25283] Prepping the Speech https://youtu.be/YXvl6X3D_nA
- [Q:25286] Words for Delivery (toy reward)
- [Q:25287] Words for Delivery (flying to the battle area)
- [Q:25393] Operation: Gnomeregan! https://youtu.be/tCm26NzjNUw

## HOW TO TEST THE CHANGES:

1) .event start 82
2) Take https://www.wowhead.com/quest=25229/a-few-good-gnomes from Ironforge and do the chain

## KNOWN ISSUES AND TODO LIST:

- [x] Fix the build
- [ ] Make sure everything works as intended and can be repeated (phasing, queueing, zoning..)
- [ ] Ensure that the texts are 100% blizlike, once done - port to db
- [ ] Camera-like vehicle that the character jumps on.. (the ending cinematic)

(for which we need sniffs or HD videos to script them off from)

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by FALL1N1** [comment]
still a WIP, so do not merge, ty

File: src/server/scripts/EasternKingdoms/OperationGnomeregan/operation_gnomeregan.cpp [hunk 1]
```diff
@@ -0,0 +1,2566 @@
...
+                        break;
+                    case 67:
+                        SetHoldState(true);
+                        JumpToNextStep(3000);
+                        break;
+                    case 68:
+                        SetHoldState(true);
+                        JumpToNextStep(3500);
+                        break;
+                    case 71:
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/OperationGnomeregan/operation_gnomeregan.cpp:843]: what is this?

File: src/server/scripts/EasternKingdoms/OperationGnomeregan/operation_gnomeregan.h [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+    STEAM_12,
+    STEAM_13,
+    STEAM_14,
+    STEAM_15,
+    STEAM_16,
+    STEAM_17,
+    STEAM_18,
+};
+
+#define MEK_1_1                 "Citizens and Friends of Gnomeregan!"
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/OperationGnomeregan/operation_gnomeregan.h:426]: all of  this should be written in proper creature_text/gossip/whatever entries and not hardcoded

File: src/server/scripts/EasternKingdoms/OperationGnomeregan/operation_gnomeregan.cpp [hunk 3]
```diff
@@ -0,0 +1,2566 @@
...
+class npc_shoot_bunny : public CreatureScript
+{
+    public:
+        npc_shoot_bunny() : CreatureScript("npc_shoot_bunny") { }
+
+        struct npc_shoot_bunnyAI : public ScriptedAI
+        {
+            npc_shoot_bunnyAI(Creature* creature) : ScriptedAI(creature) { }
+
+        void Reset()
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/OperationGnomeregan/operation_gnomeregan.cpp:2528]: code style here is broken

**Comment by Nyeriah** [comment]
Hey,

Don't mind me asking, but did you write this yourself or did you cherry-pick it from somewhere?

File: data/sql/updates/pending_db_world/rev_1647094527133522600.sql [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+
+-- [Q:25285] In and Out
+UPDATE `creature_template` SET `npcflag` = `npcflag`|16777216 WHERE `entry` = 39715;
+DELETE FROM `npc_spellclick_spells` WHERE `npc_entry` = 39715;
+INSERT INTO `npc_spellclick_spells` (`npc_entry`,`spell_id`,`cast_flags`,`user_type`) VALUES
+(39715,74204,1,0);
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId` = 18 AND `SourceGroup` = 39715;
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`SourceId`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionTarget`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`NegativeCondition`,`ErrorType`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
+(18,39715,74204,0,0,9,0,25285,0,0,0,0,0,"","Spellclick 'Summon Tank' requires quest 'In and Out' active");
+UPDATE `creature_template` SET `spell1` = 74153, `spell2` = 0, `spell3` = 0, `spell4` = 0 WHERE `entry` = 39682;
```
Comment (acidmanifesto) [inline data/sql/updates/pending_db_world/rev_1647094527133522600.sql:79]: We do not have a spell1, spell2, spell3, spell4 collumn

**Review by acidmanifesto — CHANGES_REQUESTED** [review]

**Comment by FALL1N1** [comment]
Guys, please do not comment on the code yet, this is still a WIP and a backport of the old PR here https://github.com/azerothcore/azerothcore-wotlk/pull/3678

A lot of things has to be rewritten - I know, I am working on it as we speak.


As of the original code - no, it is not known, if the author recognizes himself - feel free to add appropriate credits.
I am rewriting it to the AC standards as well as adding misc data like gobjects, npcs etc that I managed to scrap out of some sniffs.

**Comment by acidmanifesto** [comment]
> Guys, please do not comment on the code yet, this is still a WIP and a backport of the old PR here #3678
> 
> A lot of things has to be rewritten - I know, I am working on it as we speak.
> 
> As of the original code - no, it is not known, if the author recognizes himself - feel free to add appropriate credits. I am rewriting it to the AC standards as well as adding misc data like gobjects, npcs etc that I managed to scrap out of some sniffs.
> 
> I will let you know once everything is ready to be reviewed or tested.

I am not trying to sound harsh on the matter. but ensure it is worked out on your local before pushing to a pr, as we got run times that could be used for other critical things at the moment.

**Comment by Nyeriah** [comment]
> Guys, please do not comment on the code yet, this is still a WIP and a backport of the old PR here #3678
> 
> A lot of things has to be rewritten - I know, I am working on it as we speak.
> 
> As of the original code - no, it is not known, if the author recognizes himself - feel free to add appropriate credits. I am rewriting it to the AC standards as well as adding misc data like gobjects, npcs etc that I managed to scrap out of some sniffs.
> 
> I will let you know once everything is ready to be reviewed or tested.

Thanks for the answer. Could you tell us where you got this code from? There were incidents in the past of repositories being taken down by DMCA claims so we should be extra careful when cherry-picking code from closed sources, so we don't end up having issues ourselves. Thanks for your contribution though.

**Comment by FALL1N1** [comment]
@acidmanifesto All good, do not worry.  I push changes only when it's ready to be tested, I'm quite aware.

@Nyeriah the code was from the comments on the original issue, although the comment seems no longer available. After a quick search I managed to find some example ones:
https://github.com/Desch/MythCore
https://github.com/mynew4/pvelich
https://github.com/artkeep/trinity

None of which are the original author, it's publicly available on GitHub as well as BitBucket so I doubt we'd get DMCA because of it.

**Comment by acidmanifesto** [comment]
> @acidmanifesto All good, do not worry. I push changes only when it's ready to be tested, I'm quite aware. The PR is just a placeholder so anyone with sniffs/information about the event itself can share and eventually get it as close to blizzlike as possible.
> 
> @Nyeriah the code was from the comments on the original issue, although the comment seems no longer available. After a quick search I managed to find some example ones: https://github.com/Desch/MythCore https://github.com/mynew4/pvelich https://github.com/artkeep/trinity
> 
> None of which are the original author, it's publicly available on GitHub as well as BitBucket so I doubt we'd get DMCA because of it.

original orgins was found. am waiting for a bit of information on the particular it originated from as it was a former pserver core from 2012. 

So until i hear further if it was a authorized release or not i went ahead and edited in the body of the pr the appropiate credits 
Cherry-Picked: https://github.com/rebirth-core/Rebirth---WoW/commit/4342a1061984330d1e0687cb4e355afcc33248eb
Co-Authored-By: [PatrickFarnkopfOld](https://github.com/rebirth-core/Rebirth---WoW/commits?author=PatrickFarnkopfOld)
Co-Authored-By: [FrEdDy32](https://github.com/FrEdDy32)
Co-Authored-By:[Nebukad](https://github.com/Nebukad)
Co-Authored-By:[rbkai](https://github.com/rbkai)

**Comment by FALL1N1** [comment]
still a WIP, I will update once it's ready to be tested...

**Comment by acidmanifesto** [comment]
What is the current status of this? Is it abandoned?

**Comment by FALL1N1** [comment]
> What is the current status of this? Is it abandoned?

not really, I'm still gathering data for the last quest (gameobjects, texts..etc)

**Comment by acidmanifesto** [comment]
Its been about 22 days since last response, and close to 3 months since last movement, is this PR abandon? If not then please fetch a upstream. If not it will be marked abandoned and closed.

**Comment by FALL1N1** [comment]
> Its been about 22 days since last response, and close to 3 months since last movement, is this PR abandon? If not then please fetch a upstream. If not it will be marked abandoned and closed.

will do tomorrow, also push the rest of the things that are now fixed

**Comment by acidmanifesto** [comment]
Marked as abandon.

**Pull request closed without merging**